phpBB/assets/javascript/editor.js

```diff
@@ -226,6 +229,61 @@ function addquote(post_id, username, l_wrote) {
 	return;
 }
 
+/**
+* Create a quote block for given text
+*
+* Possible attributes:
+*   - author:  author's name (usually a username)
+*   - post_id: post_id of the post being quoted
+*   - user_id: user_id of the user being quoted
+*   - time:    timestamp of the original message
+*
+* @param  {!string} text       Quote's text
+* @param  {!Object} attributes Quote's attributes
+* @return {!string}            Quote block to be used in a new post/text
```

Comment:
I Googled "JSDoc type annotations" and got to this page: http://usejsdoc.org/tags-type.html

That looks like it. Do you want me to remove them or?